solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClient.java

```diff
@@ -374,7 +374,7 @@ protected HttpRequestBase createMethod(SolrRequest<?> request, String collection
 
     if (request instanceof V2Request) {
       if (System.getProperty("solr.v2RealPath") == null || ((V2Request) request).isForceV2()) {
-        basePath = baseUrl.replace("/solr", "/api");
+        basePath = changeV2RequestEndpoint(baseUrl);
```

Comment:
Not sure if this needs to be updated as HttpSolrClient class is deprecated

Reply:
unfortunately, until we get `HttpSolrClient` gone, we probably should because it is used in so many places.  We really DO need to get the migration done in Solr 10 so we can cut down on the surface of changes every time we touch an implementation.